src/EnergyPlus/OutputProcessor.cc

```diff
@@ -302,163 +302,49 @@ namespace OutputProcessor {
-        int Pos;
-        int MinLook;
-        int MaxLook;
 
         // Make sure that input has been read
         GetReportVariableInput(state);
 
         auto &op(state.dataOutputProcessor);
 
-        if (op->NumOfReqVariables > 0) {
-            // Do a quick check
-            Item = UtilityRoutines::FindItem(VarName, op->ReqRepVars, &ReqReportVariables::VarName);
-
-            op->NumExtraVars = 0;
-            op->ReportList = 0;
-            MinLook = 999999999;
-            MaxLook = -999999999;
-
-            if (Item != 0) {
-                Loop = Item;
-                Pos = Item;
-                MinLook = min(MinLook, Pos);
-                MaxLook = max(MaxLook, Pos);
-                while (Loop <= op->NumOfReqVariables && Pos != 0) {
-                    //  Mark all with blank keys as used
-                    if (op->ReqRepVars(Loop).Key.empty()) {
-                        op->ReqRepVars(Loop).Used = true;
-                    }
-                    if (Loop < op->NumOfReqVariables) {
-                        Pos = UtilityRoutines::FindItem(VarName, op->ReqRepVars({Loop + 1, op->NumOfReqVariables}), &ReqReportVariables::VarName);
-                        if (Pos != 0) {
-                            MinLook = min(MinLook, Loop + Pos);
-                            MaxLook = max(MaxLook, Loop + Pos);
-                        }
-                    } else {
-                        Pos = 1;
-                    }
-                    Loop += Pos;
-                }
-                BuildKeyVarList(state, KeyedValue, VarName, MinLook, MaxLook);
-                AddBlankKeys(state, VarName, MinLook, MaxLook);
-            }
-        }
-    }
-
-    void BuildKeyVarList(EnergyPlusData &state,
-                         std::string const &KeyedValue,   // Associated Key for this variable
-                         std::string const &VariableName, // String Name of variable
-                         int const MinIndx,               // Min number (from previous routine) for this variable
-                         int const MaxIndx                // Max number (from previous routine) for this variable
-    )
-    {
-
-        // SUBROUTINE INFORMATION:
-        //       AUTHOR         Linda K. Lawrie
-        //       DATE WRITTEN   March 1999
-        //       MODIFIED       na
-        //       RE-ENGINEERED  na
-
-        // PURPOSE OF THIS SUBROUTINE:
-        // This subroutine builds an initial list (from ReqRepVars) of
-        // pointers to that data structure for this KeyedValue and VariableName.
+        // Zero out the array / counter we use to determine if there are duplicates
+        op->NumExtraVars = 0;
+        op->ReportList = 0;
 
-        // METHODOLOGY EMPLOYED:
-        // Go through the ReqRepVars list and add those
-        // that match (and dont duplicate ones already in the list).
+        for (int i = 1; i <= op->NumOfReqVariables; ++i) {
+            auto &reqRepVar = op->ReqRepVars(i);
```

Comment:
@amirroth 's comment on a commit (which is much harder to see / find) here: https://github.com/NREL/EnergyPlus/commit/bd6ce055274872a4571b4bfcdda0fb5e588c3950#r78895379

> 📌 Most EnergyPlus uses of `std::map` or `std::unordered_map` are misguided, i.e., a map is overkill because the expected number of elements is small and/or the elements are compile time constants which maps cannot be. Then there are situations like this which look like good fits for a map--potentially large number of dynamic items--and we are rolling with an `Array1D`.